packages/notebook/src/widget.ts

```diff
@@ -1274,6 +1282,32 @@ export class Notebook extends StaticNotebook {
     this.addFooter();
   }
 
+  /**
+   * Access previous item in ipython kernel for current kernel
+   */
+  accessLastHistory(): void {
```

Comment:
I am worried about this being added on `Notebook` class - it extends the public API by two non-optional methods. What if instead the implementation was moved to respective `NotebookActions` functions (which currently just call these methods) and `history` getter were made public as an optional member on `Notebook`? This way is also easier to preserver backward compatibility. In this scenario we would review `INotebookHistory` to make it as narrow as possible.